### PR TITLE
refactor: fix gas calculations and make naming consistent

### LINKED ISSUE #446: ️🛠 - Fix logic for max fee calculation for Ledger EVM transactions
### Task description

The max fees that show in the app and on the Ledger device are not the same.

### Requirements

_TBD_

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/src/lib/core/layer-2/actions/getGasPriceInWei.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    const chain = getNetwork()?.getChain(chainId)
+
+    const provider = chain?.getProvider()
+    if (!provider) {
+        return undefined
+    }
+
+    const gasPrice = await provider.eth.getGasPrice()
+    const hexGasPrice = Converter.decimalToHex(Number(gasPrice), true)
+    return hexGasPrice
```
Comment (maxwellmattryan) [inline packages/shared/src/lib/core/layer-2/actions/getGasPriceInWei.ts]: These lines can be combined

File: packages/shared/src/lib/core/layer-2/utils/calculateGasFeeInGlow.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { BigIntLike } from '@ethereumjs/util'
+import { WEI_PER_GLOW } from '../constants'
+
+export function calculateGasFeeInGlow(gasAmount: BigIntLike, gasFeeInWei: BigIntLike): string {
+    const gasPriceInGlow = BigInt(Number(gasFeeInWei)) / WEI_PER_GLOW
```
Comment (maxwellmattryan) [inline packages/shared/src/lib/core/layer-2/utils/calculateGasFeeInGlow.ts]: If this is dividing the `gasFeeInWei` by the `WEI_PER_GLOW`, wouldn't that be the `gasFeeInGlow`, not `gasPriceInGlow`? That would then mean that this function is actually returning the `gasPriceInGlow` right?

Comment (maxwellmattryan) [reply packages/shared/src/lib/core/layer-2/utils/calculateGasFeeInGlow.ts]: Ah the problem is that `gasFeeInWei` should actually be `gasPriceInWei`

File: packages/shared/src/lib/core/layer-2/utils/calculateMaxGasFeeFromTransactionData.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { EvmTransactionData } from '../types'
+import { calculateGasFeeInGlow } from './calculateGasFeeInGlow'
+
+export function calculateMaxGasFeeFromTransactionData(transactionData: EvmTransactionData): string | undefined {
```
Comment (maxwellmattryan) [inline packages/shared/src/lib/core/layer-2/utils/calculateMaxGasFeeFromTransactionData.ts]: Probably best if we start using `bigint`?

File: packages/shared/src/lib/core/layer-2/utils/getLayer2MetadataForTransfer.ts [hunk 4]
```diff
@@ -12,11 +12,13 @@ export async function getLayer2MetadataForTransfer(sendFlowParameters: SendFlowP
     const encodedAddress = address ? encodeAddress(address.toLowerCase()) : ''
 
     const estimatedGas = await estimateGasForLayer1ToLayer2Transaction(sendFlowParameters as TokenSendFlowParameters)
+    const gasLimit = estimatedGas * GAS_MULTIPLIER
```
Comment (maxwellmattryan) [inline packages/shared/src/lib/core/layer-2/utils/getLayer2MetadataForTransfer.ts]: Maybe `GAS_LIMIT_MULTIPLIER` is a better name?

File: packages/shared/src/lib/core/layer-2/utils/signEvmTransactionWithStronghold.ts [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
 import { IAccountState } from '@core/account'
-import { EvmTransactionData } from '@core/layer-2/types'
 import { prepareEvmTransaction } from '@core/layer-2/utils'
 import { Transaction } from '@ethereumjs/tx'
 import { fromRpcSig, ECDSASignature } from '@ethereumjs/util'
 import { DEFAULT_EVM_TRANSACTION_OPTIONS } from '@core/layer-2'
 import type { Bip44 } from '@iota/wallet/types'
 
 export async function signEvmTransactionWithStronghold(
-    txData: EvmTransactionData,
```
Comment (maxwellmattryan) [inline packages/shared/src/lib/core/layer-2/utils/signEvmTransactionWithStronghold.ts:10]: We should still be able to use this type no?

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Have some requested changes :pray:

File: packages/shared/src/lib/core/layer-2/utils/signEvmTransactionWithStronghold.ts (hunk 5 above)
Comment (nicole-obrien) [reply packages/shared/src/lib/core/layer-2/utils/signEvmTransactionWithStronghold.ts:10]: We extended the TxData in EvmTransactionData to include the estimatedGas and therefore this function is only expecting the raw TxData

**Review by maxwellmattryan — APPROVED** [review]

**Pull request merged**